core/hyphenator-liang.lua

```diff
@@ -114,18 +114,44 @@ local hyphenateNode = function (node)
     local hyphen = SILE.shaper:createNnodes(SILE.settings:get("font.hyphenchar"), node.options)
     local newnodes = {}
     for j, segment in ipairs(segments) do
-      local leadingApostrophe
+      local specificDiscretionary
       if segment == "" then
         SU.dump({ j, segments })
         SU.error("No hyphenation segment should ever be empty", true)
       end
       if node.options.language == "tr" then
```

Comment:
Noted in passing, the language is in `node.language` (tracked back to `shaper:formNnode`) and in `node.options.language`
It needs to be in the options (for `createNnodes` etc.). We however have two possibilities and we use one or the other at different places. That's code smell -> I would suggest a dedicated refactoring issue...

Reply:
Agreed, that doesn't pass a smell test.

Reply:
Tracked in #1965